### ISSUE TITLE
Update the ssp-demo with necessary component definition and rules

### ISSUE THREAD

**Issue by fsuits** [oscal-compass/compliance-trestle#1266]
## Issue description / feature objectives
The current ssp-demo doesn't represent the new way to generate an ssp based on component definitions and rules.  The demo needs to be heavily reworked with new content in order to serve as a guide for users of the component and ssp authoring tools.

**Comment by kmoragas-pi** [comment]
Yes, please; from the user perspective has been very difficult to run this demo. I had to create a temporary component definition in order to run it. 

On the other hand, is confusing having two ssp-demos:
- https://github.com/IBM/compliance-trestle-ssp-demo
- https://github.com/IBM/compliance-trestle-demos/tree/develop/ssp_author_demo (this one is using trestle 1.0.0)

**Comment by sunstonesecure-robert** [comment]
can you elaborate on what the general idea is so I can break it down into specific TODOs? for instance is the idea to be able to generate a complete SSP using a collection of trestle created CDefs? If so:

- I can create components using the cli `trestle create -t component-definition` but how do I add control-implementations via cli?
  - and for each c-i how do I note specific implemented-requirements in the CDef? 
  - how can I define capabilities and related c-is and i-rs?

**Comment by fsuits** [comment]
Hi @sunstonesecure-robert and @kmoragas-pi the main new complexity is how rules are defined and applied in a relevant use case.  On our side we will aim to create a meaningful and small component definition that has rules and rule parameters that are representative of expected use cases.

The previous demo was much simpler in not having this additional complexity - along with the fact that all control markdown was generated, as opposed to only those where rules apply.

As for adding control-implementations via cli, that level of detail isn't handled via markdown and normally it would already be defined in the json file for the component definition.  But trestle is intended to serve as a toolbox for breaking down a model into constituent parts and adding/editing new parts - which can be done with the split and merge tools.  Thus far those tools haven't had much usage but they are there for that purpose.

I think that in realistic use cases the component definitions will be generated by some kind of conversion from a spreadsheet to json - which then allows markdown to handle the details of prose, rule parameter values, and implementation status.

In summary, for the demo we will first create a small but representative component definition with associated rules and rule parameters - and then we will process it into a formatted ssp document.
